server/src/main/java/com/epam/aidial/core/server/controller/RouteController.java

```diff
@@ -254,20 +274,9 @@ private Route selectRoute() {
     @SneakyThrows
     private String getEndpointUri(Upstream upstream) {
         URIBuilder uriBuilder = new URIBuilder(upstream.getEndpoint());
-        if (context.isRewritePath()) {
+        if (context.getRoute() != null && context.getRoute().isRewritePath()) {
```

Comment:
`context.getRoute()` might return null?

Reply:
yes, must not be null, removed this condition